libtiledbsoma/src/utils/fastercsx.h

```diff
@@ -0,0 +1,545 @@
+            throw std::out_of_range("Coordinate out of range.");
+
+        Bj[dest] = Aj_[n];
+        Bd[dest] = Ad_[n];
+    }
+}
+
+/**
+ * Given COO matrix in <I,J,D> format (row, col, value), compress into CSX
+ * <P,J,D>. To generate a CSC, just swap the major/minor axes.
+ *
+ * Caller must allocate output buffers, but need not initialize them.
+ *
+ * @param tp ThreadPool
+ * @param n_row Size of matrix on major axis
+ * @param n_col Size of matrix on minor axis
+ * @param nnz Number of values
+ * @param Ai Input, major axis coordinates
+ * @param Aj Input, minor axis coordinates
+ * @param Ad Input values
+ * @param Bp Output, compressed major axis.
+ * @param Bj Output, minor axis coordinates.
+ * @param Bd Output values
+ */
+template <class VALUE, class COO_IDX, class CSX_MINOR_IDX, class CSX_MAJOR_IDX>
+void compress_coo(
+    ThreadPool* const tp,
+    const Shape& shape,
+    uint64_t nnz,
+    const std::vector<tcb::span<const COO_IDX>>& Ai,
+    const std::vector<tcb::span<const COO_IDX>>& Aj,
+    const std::vector<tcb::span<const VALUE>>& Ad,
+    tcb::span<CSX_MAJOR_IDX> Bp,
+    tcb::span<CSX_MINOR_IDX> Bj,
+    tcb::span<VALUE> Bd) {
+    auto [n_row, n_col] = shape;
+    assert(Ai.size() == Aj.size() && Aj.size() == Ad.size());
+    assert(sum_over_size_(Ai) == nnz);
+    assert(sum_over_size_(Aj) == nnz);
+    assert(sum_over_size_(Ad) == nnz);
+    assert(Bp.size() == n_row + 1);
+    assert(Bj.size() == nnz);
+    assert(Bd.size() == nnz);
+
+    // Get major axis counts. Important side effect: range checks the Ai
+    // values.
+    count_rows_(tp, n_row, nnz, Ai, Bp);
+    sum_rows_to_pointers_(Bp);
+    assert(Bp[n_row] >= 0 && static_cast<uint64_t>(Bp[n_row]) == nnz);
+
+    std::vector<CSX_MAJOR_IDX> Bp_left(Bp.begin(), Bp.end() - 1);
+    std::vector<CSX_MAJOR_IDX> Bp_right(Bp.begin() + 1, Bp.end());
+    tcb::span<CSX_MAJOR_IDX> Bp_left_span{Bp_left};
+    tcb::span<CSX_MAJOR_IDX> Bp_right_span{Bp_right};
+
+    // Parallel, lock-free copy of all minor index and data values. Partitioned
+    // by contiguous major dimension (Ai) values, minimizing the write
+    // contention in the indptr/Bp updates. Each partition is split into two
+    // parts, which can run in parallel, by working either left or right (minor
+    // dimension beginning and end of range) - this reduces the number of major
+    // dimension reads by half.
+    //
+    const auto partition_bits = std::max(
+                                    13L,
+                                    std::lround(std::ceil(std::log2(
+                                        n_row / tp->concurrency_level())))) +
+                                1;
+    const auto n_partitions = (n_row + (1 << partition_bits) - 1) >>
+                              partition_bits;
+    assert((n_partitions > 0) || (n_row == 0));
+    auto status = parallel_for(
+        tp,
+        0ul,
+        2 * n_partitions,
+        [&n_partitions,
+         &partition_bits,
```

Comment:
```suggestion
        [&partition_bits,
```

Building locally on MacOS I get a warning that `n_partitions` is unused within the lambda body (which is correct).

Reply:
oh, easy fix - I just missed this one